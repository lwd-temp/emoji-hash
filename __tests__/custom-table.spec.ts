import Hasher, { getHash, useTable } from '../src';
import fakeTable from '../__mocks__/table';

describe('Usage with different table', () => {
  const originalTable = Hasher.table;
  const testTableForHash = [
    'https://yandex.ru',
    'https://github.com/opa-oz/emoji-hash',
    'My text is awesome',
    'Hi guys',
    'opa-oz',
    '@gahara',
    'Python',
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book"
  ];

  beforeAll(() => {
    useTable(fakeTable);
  });

  afterAll(() => {
    useTable(originalTable);
  });

  testTableForHash.forEach((input) => {
    test('should generate hash correctly with fake table', () => {
      expect(getHash(input)).toMatchSnapshot();
    });
  });
});
