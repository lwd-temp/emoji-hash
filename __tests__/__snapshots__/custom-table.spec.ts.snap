// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Usage with different table should generate hash correctly with fake table 1`] = `"🈴🍟😟🌐🌴🆚"`;

exports[`Usage with different table should generate hash correctly with fake table 2`] = `"🚺🍅🍬📝💘🐽"`;

exports[`Usage with different table should generate hash correctly with fake table 3`] = `"🚺🍬🌉🐰♐️🔄"`;

exports[`Usage with different table should generate hash correctly with fake table 4`] = `"🚺🎌🚩📺🐬💳👑"`;

exports[`Usage with different table should generate hash correctly with fake table 5`] = `"🚺🎌🎩🚆😔🎌🎩"`;

exports[`Usage with different table should generate hash correctly with fake table 6`] = `"🚺💹🚳🔳🎌🍬"`;

exports[`Usage with different table should generate hash correctly with fake table 7`] = `"🚺🈴😔💳🍆🎌💘"`;

exports[`Usage with different table should generate hash correctly with fake table 8`] = `"🚺🎌📝🕕🈚️🚆📺"`;
